openedx/core/djangoapps/course_groups/tests/test_views.py

```diff
@@ -92,49 +93,114 @@ def _verify_non_staff_cannot_access(self, view, request_method, view_args):
         view_args.insert(0, request)
         self.assertRaises(Http404, view, *view_args)
 
-
-class CohortHandlerTestCase(CohortViewsTestCase):
-    """
-    Tests the `cohort_handler` view.
-    """
-    def get_cohort_handler(self, course, cohort=None):
+    def get_handler(self, course, cohort=None, expected_response_code=200, handler=cohort_handler):
         """
-        Call a GET on `cohort_handler` for a given `course` and return its response as a
-        dict. If `cohort` is specified, only information for that specific cohort is returned.
+        Call a GET on `handler` for a given `course` and return its response as a dict.
+        Raise an exception if response status code is not as expected.
         """
         request = RequestFactory().get("dummy_url")
         request.user = self.staff_user
         if cohort:
-            response = cohort_handler(request, unicode(course.id), cohort.id)
+            response = handler(request, unicode(course.id), cohort.id)
         else:
-            response = cohort_handler(request, unicode(course.id))
-        self.assertEqual(response.status_code, 200)
+            response = handler(request, unicode(course.id))
+        self.assertEqual(response.status_code, expected_response_code)
         return json.loads(response.content)
 
-    def put_cohort_handler(self, course, cohort=None, data=None, expected_response_code=200):
+    def put_handler(self, course, cohort=None, data=None, expected_response_code=200, handler=cohort_handler):
         """
-        Call a PUT on `cohort_handler` for a given `course` and return its response as a
-        dict. If `cohort` is not specified, a new cohort is created. If `cohort` is specified,
-        the existing cohort is updated.
+        Call a PUT on `handler` for a given `course` and return its response as a dict.
+        Raise an exception if response status code is not as expected.
         """
         if not isinstance(data, basestring):
             data = json.dumps(data or {})
         request = RequestFactory().put(path="dummy path", data=data, content_type="application/json")
         request.user = self.staff_user
         if cohort:
-            response = cohort_handler(request, unicode(course.id), cohort.id)
+            response = handler(request, unicode(course.id), cohort.id)
         else:
-            response = cohort_handler(request, unicode(course.id))
+            response = handler(request, unicode(course.id))
         self.assertEqual(response.status_code, expected_response_code)
         return json.loads(response.content)
 
+
+class CourseCohortSettingsHandlerTestCase(CohortViewsTestCase):
+    """
+    Tests the `course_cohort_settings_handler` view.
+    """
+    def test_non_staff(self):
+        """
+        Verify that we cannot access course_cohort_settings_handler if we're a non-staff user.
```

Comment:
Just to be clear, I think this is 'non-course staff', right? The only people who can modify the cohort setting are course staff or global staff (aka admins).